src/cluster/kv/etcd/store.go

```diff
@@ -616,7 +617,32 @@ func (c *client) writeCacheToFile() error {
 	return nil
 }
 
+func (c *client) createCacheDir() error {
+	path := path.Dir(c.opts.CacheFileFn()(c.opts.Prefix()))
```

Comment:
I think `os.MkdirAll` is what you're looking for here.

This might be helpful as an example: https://github.com/m3db/m3/blob/master/src/dbnode/server/server.go#L188

Reply:
Ah good point with `os.MkdirAll`, thanks for the hint. 

I tried using `lockfile.CreateAndAcquire` but somehow the other tests are failing then:

```
--- FAIL: TestCache (0.70s)
        Error Trace:    store_test.go:198
        Error:          Received unexpected error "grpc: the client connection is closing"

--- FAIL: TestStaleDelete__FromGet (61.54s)
        Error Trace:    store_test.go:832
        Error:          Should be true
        Messages:       timed out waiting for client to read values from cache

--- FAIL: TestStaleDelete__FromWatch (62.06s)
        Error Trace:    store_test.go:917
        Error:          Should be true
        Messages:       timed out waiting for client to read values from cache
```

I'll just leave it with `os.MkDirAll` for now, but happy to take advice on how to properly implement the fs lock, if required.

Reply:
Sorry I wasn't suggesting that you use `lockfile.CreateAndAcquire` I was just pointing to its implementation which uses os.MkdirAll (and passes the right permissions as well)